driver-core/src/main/com/mongodb/internal/connection/LevelCountingBsonWriter.java

```diff
@@ -16,270 +16,40 @@
 
 package com.mongodb.internal.connection;
 
-import org.bson.BsonBinary;
 import org.bson.BsonBinaryWriter;
-import org.bson.BsonDbPointer;
-import org.bson.BsonReader;
-import org.bson.BsonRegularExpression;
-import org.bson.BsonTimestamp;
 import org.bson.BsonWriter;
-import org.bson.types.Decimal128;
-import org.bson.types.ObjectId;
 
-import static org.bson.assertions.Assertions.notNull;
 
-
-abstract class LevelCountingBsonWriter implements BsonWriter {
-    private final BsonBinaryWriter bsonBinaryWriter;
+abstract class LevelCountingBsonWriter extends BsonWriterAdapter implements BsonWriter {
```

Comment:
As `BsonWriterAdapter` implements `BsonWriter` does it also need to be declared here?

Reply:
It doesn't.  Good catch, I removed it.